src/Bicep.Core/Emit/InlineDependencyVisitor.cs

```diff
@@ -261,17 +272,28 @@ private void VisitFunctionCallSyntaxBaseInternal(FunctionCallSyntaxBase syntax)
             switch (model.GetSymbolInfo(syntax))
             {
                 case FunctionSymbol functionSymbol:
-                    SetCache(shouldInlineCache[currentDeclaration] || functionSymbol.FunctionFlags.HasFlag(FunctionFlags.RequiresInlining));
+                    SetInlineCache(shouldInlineCache[currentDeclaration] || functionSymbol.FunctionFlags.HasFlag(FunctionFlags.RequiresInlining));
                     return;
             }
         }
 
-        private void SetCache(bool shouldInline)
+        private void SetInlineCache(bool shouldInline)
         {
             if (this.currentDeclaration is not null)
             {
                 this.shouldInlineCache[this.currentDeclaration] = shouldInline;
             }
         }
+        private void SetNotInlineCache(bool shouldNotInline)
+        {
+            if (this.currentDeclaration is not null)
+            {
+                this.shouldInlineCache[this.currentDeclaration] = !shouldNotInline;
+                if (shouldNotInline)
+                {
+                    this.shouldNotInlineCache.Add(this.currentDeclaration);
+                }
+            }
```

Comment:
I'm assuming this logic is to differentiate the default (don't inline) from an explicit decision not to inline. Rather than introducing 2 caches, would it be possible to modify to use a single cache with an enum - e.g. values to indicate "NotEvaluated", "Inline", "DoNotInline"?

Reply:
+1

Reply:
I added enum with `NotInline`, `Inline` and `SkipInline` values.